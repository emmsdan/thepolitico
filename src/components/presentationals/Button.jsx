import React from 'react';

const Button = props => {
  if (props.button.type === 'href') {
    return (
      <a href={props.button.href} className={props.button.class}>
        {props.button.value}
      </a>
    );
  } else {
    return (
      <button className={props.button.class} type={props.button.type}>
        {props.button.value}
      </button>
    );
  }
};
export default Button;
