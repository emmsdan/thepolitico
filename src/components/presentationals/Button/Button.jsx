import React from 'react';

const Button = ({ button }) => {
  if (button.type === 'href') {
    return (
      <a href={button.href} className={button.className}>
        {button.value}
      </a>
    );
  } else {
    return (
      <button className={button.className} type={button.type}>
        {button.value}
      </button>
    );
  }
};
export default Button;
