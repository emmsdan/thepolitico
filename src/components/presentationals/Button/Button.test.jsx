import React from 'react';
import { shallow } from 'enzyme';

import Button from './Button';

describe('Button', () => {
  const button = {
    className: 'btn-large',
    value: 'create account',
    type: 'button',
    href: 'user-url',
  };

  it('should render correctly', () => {
    const component = shallow(<Button button={button} />);
    expect(component).toMatchSnapshot();
  });

  it('should render link button correctly', () => {
    button.type = 'href';
    const component = shallow(<Button button={button} />);
    expect(component).toMatchSnapshot();
  });
});
