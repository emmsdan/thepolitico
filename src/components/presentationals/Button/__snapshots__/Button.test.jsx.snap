// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button
  className="btn-large"
  type="button"
>
  create account
</button>
`;

exports[`Button should render link button correctly 1`] = `
<a
  className="btn-large"
  href="user-url"
>
  create account
</a>
`;
